### PR TITLE
Sweets/profile

### PR DESCRIPTION
**Pull request by sweetmantech**
# User Profile
- new route: `/collector/{collectorId}`
- `collectorId` can be any valid Ethereum Address.
- future: `collectorId` can also be an ENS name.

## Examples
- https://evil-corp-m8uve77ag-defient-team.vercel.app/collector/0x849f03ACc35C6F4A861b76e1F271d217CD24b18C
- https://evil-corp-m8uve77ag-defient-team.vercel.app/collector/0x2697f24A0128F4BC44c37576152518EC6Cd70924
- https://evil-corp-m8uve77ag-defient-team.vercel.app/collector/0x135538daed6657644b7A90958183d760f508BA45

Screenshot 2023-03-22 at 9 40 47 PM

### CODE REVIEW

File: Models/AllowList copy.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import { Schema, model, models, Model } from "mongoose"
+
```
Comment (crazzywizard) [inline Models/AllowList copy.tsx]: why is this file necessary? how is it different from the existing Allowlist model?

File: components/CollectorPage/CollectorPage.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import { useRouter } from "next/router"
+import ImageCard from "./ImageCard"
+import balanceOfParticipationRewards from "../../lib/balanceOfParticipationRewards"
+import truncate from "../../lib/truncate"
+
+function CollectorPage() {
+  const { data: signer } = useSigner()
+  const router = useRouter()
+  const { collectorId } = router.query
+  const [balance, setBalance] = useState("0")
```
Comment (crazzywizard) [inline components/CollectorPage/CollectorPage.tsx]: why is this a string of zero? What does it signifiy?

File: lib/truncate.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+const truncate = (text: string) =>
+  text?.length > 10 ? `${text.substring(0, 5)}...${text.substring(text.length - 4)}` : text
```
Comment (crazzywizard) [inline lib/truncate.ts]: this is unclear, too much usage of magic numbers which make it harder to understand code a week from today.

**Review by crazzywizard — CHANGES_REQUESTED** [review]
Some changes needed, overall looks good.

File: components/CollectorPage/CollectorPage.tsx (hunk 2 above)
Comment (sweetmantech) [reply components/CollectorPage/CollectorPage.tsx]: 0 tokens owned

Comment (sweetmantech) [reply components/CollectorPage/CollectorPage.tsx]: what would you prefer it say?

Comment (crazzywizard) [reply components/CollectorPage/CollectorPage.tsx]: make a constant that says `NUMBER_OF_TOKENS` and set it to zero and use that as the initial value of balance

File: lib/truncate.ts (hunk 3 above)
Comment (sweetmantech) [reply lib/truncate.ts]: updated to remove "magic numbers" 🚮 
Screenshot 2023-03-22 at 9 58 04 PM

File: components/CollectorPage/CollectorPage.tsx (hunk 2 above)
Comment (crazzywizard) [reply components/CollectorPage/CollectorPage.tsx]: does it have to be a string? if the data is coming back as an integer then is string necessary?

File: Models/AllowList copy.tsx (hunk 1 above)
Comment (sweetmantech) [reply Models/AllowList copy.tsx]: removed 🚮

File: components/CollectorPage/CollectorPage.tsx (hunk 2 above)
Comment (sweetmantech) [reply components/CollectorPage/CollectorPage.tsx]: data is coming back as `BigNumber`

Comment (sweetmantech) [reply components/CollectorPage/CollectorPage.tsx]: added `const NUMBER_OF_TOKENS = "0"`

**Review by crazzywizard — APPROVED** [review]
LGTM! 🚢
